### PR TITLE
Improve logging, error handling, fix batch commitments

### PR DESCRIPTION
**Pull request by forgetso**
- upgrade polkadot deps
- Get historic queries working
- Update contract query method to allow for historic calls
- Update getStorage
- upgrade deps, remove contract.ts
- Fix tests
- Update contract addresses, ports to be non-default
- all tests working and newest version of contract updated in env files
- Update metadata.json
- new contract address with all tests working
- Get contract package working with new version of ink and substrate contracts node

### CODE REVIEW

File: src/abi/prosopo.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
 {
   "source": {
-    "hash": "0xad56754944fca5da205a2dbcb86b0d674e58df770bf75b862601e2812d301f7a",
+    "hash": "0xb398c34eeaa6631344073ab704806495f35c84bc73f79e69784e4dbf5ad96b02",
     "language": "ink! 4.0.0-beta",
-    "compiler": "rustc 1.68.0-nightly",
+    "compiler": "rustc 1.67.0",
```
Comment (goastler) [inline src/abi/prosopo.json:5]: are we downgrading rust?

File: src/abi/prosopo.json [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
     "build_info": {
       "build_mode": "Debug",
       "cargo_contract_version": "2.0.0-rc",
-      "rust_toolchain": "nightly-x86_64-unknown-linux-gnu",
+      "rust_toolchain": "stable-x86_64-unknown-linux-gnu",
```
Comment (goastler) [inline src/abi/prosopo.json:9]: and no need to use nightly anymore?

**Review by goastler — CHANGES_REQUESTED** [review]

File: src/abi/prosopo.json (hunk 2 above)
Comment (forgetso) [reply src/abi/prosopo.json:9]: as mentioned, I've tested with as low as 1.63. This just happened to be the toolchain installed when I did the most recent build.

File: src/abi/prosopo.json (hunk 1 above)
Comment (forgetso) [reply src/abi/prosopo.json:5]: Not intentionally, no. We need to agree a procedure for applying the most recent contract artifacts to this repo.

**Review by goastler — APPROVED** [review]

**Pull request merged**